src/Presentation/Controllers/IncomeController.cs

```diff
@@ -0,0 +1,93 @@
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using System.Threading.Tasks;
+using ExpenseTrackerGrupo4.src.Domain.Entities;
+using ExpenseTrackerGrupo4.src.Infrastructure.Interfaces;
+
+namespace ExpenseTrackerGrupo4.src.Presentation.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class IncomeController : ControllerBase
+    {
+        private readonly IIncomeService _incomeService;
+
+        public IncomeController(IIncomeService incomeService)
+        {
+            _incomeService = incomeService;
+        }
+
+        [HttpPost]
+        public async Task<IActionResult> CreateIncome([FromBody] Income income)
+        {
+            if (income == null)
+            {
+                return BadRequest("Income data is required.");
+            }
+
+            income.UserId = GetUserId();
+
+            await _incomeService.AddIncomeAsync(income);
+            return CreatedAtAction(nameof(GetIncomeById), new { id = income.Id }, income);
+        }
+
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Income>>> GetIncomes()
+        {
+            var userId = GetUserId();
+            var incomes = await _incomeService.GetIncomesByUserAsync(userId);
+            return Ok(incomes);
+        }
+
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Income>> GetIncomeById(Guid id)
+        {
+            var income = await _incomeService.GetIncomeByIdAsync(id);
+            if (income == null)
+            {
+                return NotFound();
+            }
+            return Ok(income);
+        }
+
+        [HttpPut("{id}")]
+        public async Task<IActionResult> UpdateIncome(Guid id, [FromBody] Income updatedIncome)
```

Comment:
Don't forget to validate that an income should be updated and deleted just for the user it belongs to.